tests/integration_test/http_method_override_test.go

```diff
@@ -0,0 +1,260 @@
+	t.Parallel()
+
+	s := env.NewTestEnv(comp.TestMethodOverrideBackendBody, platform.EchoSidecar)
+	defer s.TearDown()
+	if err := s.Setup(utils.CommonArgs()); err != nil {
+		t.Fatalf("fail to setup test env, %v", err)
+	}
+
+	testData := []struct {
+		desc          string
+		path          string
+		method        string
+		headers       map[string]string
+		body          string
+		wantResp      string
+		httpCallError error
+	}{
+		{
+			desc:   "Overridden PUT has body",
+			path:   "/echo?key=api-key",
+			method: "PUT",
+			headers: map[string]string{
+				"X-HTTP-Method-Override": "POST",
+			},
+			body:     `hello`,
+			wantResp: `{"message":"hello"}`,
+		},
+		{
+			desc:   "Overridden GET has no body for the backend to handle",
+			path:   "/echo?key=api-key",
+			method: "GET",
+			headers: map[string]string{
+				"X-HTTP-Method-Override": "POST",
+			},
+			body:          "This body will not be sent in the request",
+			httpCallError: fmt.Errorf(`{"code":500,"message":"Could not get body: EOF"}`),
+		},
+	}
+	for _, tc := range testData {
+		url := fmt.Sprintf("http://localhost:%v%v", s.Ports().ListenerPort, tc.path)
+		gotResp, err := client.DoWithHeaders(url, tc.method, tc.body, tc.headers)
+
+		if tc.httpCallError == nil {
+			if err != nil {
+				t.Fatal(err)
+			}
+		} else {
+			if !strings.Contains(err.Error(), tc.httpCallError.Error()) {
+				t.Errorf("expected Http call error: %v, got: %v", tc.httpCallError, err)
+			}
+			continue
+		}
+		gotRespStr := string(gotResp)
+		if err := util.JsonEqual(tc.wantResp, gotRespStr); err != nil {
+			t.Errorf("Test(%s) fails: \n %s", tc.desc, err)
+		}
+	}
+}
+
+func TestMethodOverrideScReport(t *testing.T) {
+	t.Parallel()
+
+	s := env.NewTestEnv(comp.TestMethodOverrideScReport, platform.EchoSidecar)
+	s.AppendHttpRules([]*annotationspb.HttpRule{
+		{
+			Selector: "1.echo_api_endpoints_cloudesf_testing_cloud_goog.Echo_nokey_override_as_get",
+			Pattern: &annotationspb.HttpRule_Get{
+				Get: "/echo/nokey/OverrideAsGet",
+			},
+		},
+	})
+	s.AppendUsageRules(
+		[]*confpb.UsageRule{
+			{
+				Selector:               "1.echo_api_endpoints_cloudesf_testing_cloud_goog.Echo_nokey_override_as_get",
+				AllowUnregisteredCalls: true,
+			},
+		})
+
+	defer s.TearDown()
+	if err := s.Setup(utils.CommonArgs()); err != nil {
+		t.Fatalf("fail to setup test env, %v", err)
+	}
+
+	testData := []struct {
+		desc           string
+		url            string
+		method         string
+		requestHeader  map[string]string
+		message        string
+		wantResp       string
+		httpCallError  error
+		wantScRequests []interface{}
+	}{
+		{
+			desc:    "Overridden POST is displayed as GET in SC Report",
+			url:     fmt.Sprintf("http://localhost:%v%v", s.Ports().ListenerPort, "/echo/nokey/OverrideAsGet"),
+			message: "hello hello",
+			method:  "POST",
+			requestHeader: map[string]string{
+				"X-HTTP-Method-Override": "GET",
+			},
+			httpCallError: fmt.Errorf("405 Method Not Allowed"),
```

Comment:
where is this 405 replied,  by echo backend server? 
can we have a case with a success one?

Reply:
Good idea, it's confusing. Replaced with a success case only.